forest/daemon/src/daemon.rs

```diff
@@ -69,24 +71,25 @@ pub(super) async fn start(config: Config, detached: bool) -> Db {
     );
 
     let path: PathBuf = config.client.data_dir.join("libp2p");
-    let net_keypair = get_keypair(&path.join("keypair")).unwrap_or_else(|| {
-        // Keypair not found, generate and save generated keypair
-        let gen_keypair = ed25519::Keypair::generate();
-        // Save Ed25519 keypair to file
-        // TODO rename old file to keypair.old(?)
-        match write_to_file(&gen_keypair.encode(), &path, "keypair") {
-            Ok(file) => {
-                // Restrict permissions on files containing private keys
-                #[cfg(unix)]
-                forest_utils::io::set_user_perm(&file).expect("Set user perms on unix systems");
-            }
-            Err(e) => {
-                info!("Could not write keystore to disk!");
-                trace!("Error {:?}", e);
-            }
-        };
-        Keypair::Ed25519(gen_keypair)
-    });
+    let net_keypair = match get_keypair(&path.join("keypair")) {
+        Some(keypair) => Ok::<forest_libp2p::Keypair, std::io::Error>(keypair),
+        None => {
+            let gen_keypair = ed25519::Keypair::generate();
+            // Save Ed25519 keypair to file
+            // TODO rename old file to keypair.old(?)
+            match write_to_file(&gen_keypair.encode(), &path, "keypair") {
+                Ok(file) => {
+                    // Restrict permissions on files containing private keys
+                    #[cfg(unix)]
+                    forest_utils::io::set_user_perm(&file)?;
+                }
+                Err(e) => {
+                    anyhow::bail!("Could not write keystore to disk! Error: {e}");
+                }
+            };
```

Comment:
Let's modify `set_user_perm` such that it is always defined but may be a no-op.

```suggestion
            let file = write_to_file(&gen_keypair.encode(), &path, "keypair")?;
            // Restrict permissions on files containing private keys
            forest_utils::io::set_user_perm(&file)?;
```

Reply:
Done